src/main/java/com/google/api/gax/core/NanoClock.java

```diff
@@ -29,23 +29,33 @@
  */
 package com.google.api.gax.core;
 
-/**
- * An interface for getting the current value of a high-resolution time source, in nanoseconds.
- *
- * Clocks other than DefaultNanoClock are typically used only for testing.
- *
- * This interface is required in addition to Java 8's Clock, because nanoTime is required to compare
- * values with io.grpc.CallOptions.getDeadlineNanoTime().
- */
-public interface NanoClock {
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+/** Default implementation of the ApiClock interface, using call to System.nanoTime(). */
+public final class NanoClock implements ApiClock, Serializable {
```

Comment:
But... this one calls `System.nanoTime()`... That same confusion is present on the `System` class itself. I prefer to channel the underlying concepts in this case instead of trying to "improve" them.